fud2/src/lib.rs

```diff
@@ -458,4 +462,87 @@ pub fn build_driver(bld: &mut DriverBuilder) {
             Ok(())
         },
     );
+
+    let yxi = bld.state("yxi", &["yxi"]);
+    bld.op(
+        "calyx-to-yxi",
+        &[calyx_setup],
+        calyx,
+        yxi,
+        |e, input, output| {
+            e.build_cmd(&[output], "calyx", &[input], &[])?;
+            e.arg("backend", "yxi")?;
+            Ok(())
+        },
+    );
+
+    let wrapper_setup = bld.setup("YXI and AXI generation", |e| {
+        // Define a `gen-axi` rule that invokes our Python code generator program.
+        // For now point to standalone axi-generator.py. Can maybe turn this into a rsrc file?
+        e.config_var_or(
+            "axi-generator",
+            "axi.generator",
+            "$calyx-base/yxi/axi-calyx/axi-generator.py",
+        )?;
+        e.config_var_or("python", "python", "python3")?;
+        e.rule("gen-axi", "$python $axi-generator $in > $out")?;
+
+        // Define a simple `combine` rule that just concatenates any numer of files.
+        e.rule("combine", "cat $in > $out")?;
+
+        e.rule(
+            "remove-imports",
+            "sed '1,/component main/{/component main/!d}' $in > $out",
+        )?;
+        Ok(())
+    });
+    bld.op(
+        "axi-wrapped",
+        &[calyx_setup, wrapper_setup],
+        calyx,
+        calyx,
+        |e, input, output| {
+            // Generate the YXI file.
+            //no extension
+            let file_name = input
+                .rsplit_once('/')
+                .unwrap()
+                .1
+                .rsplit_once('.')
+                .unwrap()
+                .0;
+            let tmp_yxi = format!("{}.yxi", file_name);
+
+            //Get yxi file from main compute program.
+            //TODO(nate): Can this use the `yxi` operation instead of hardcoding the build cmd calyx rule with arguments?
```

Comment:
I think this was the only thing I wasn't sure about w.r.t best fud2 practices @sampsyo. Going to merge for now but can change this in a future PR.